syft/pkg/cataloger/rpmdb/parse_rpmdb.go

```diff
@@ -74,6 +76,19 @@ func parseRpmDB(resolver source.FilePathResolver, dbLocation source.Location, re
 	return allPkgs, nil
 }
 
+// The RPM naming scheme is [name]-[version]-[release]-[arch], where version is implicitly expands to [epoch]:[version].
+// RPM version comparison depends on comparing at least the version and release fields together as a subset of the
+// naming scheme. This toElVersion function takes a RPM DB package information and converts it into a minimally comparable
+// version string, containing epoch (optional), version, and release information. Epoch is an optional field and can be
+// assumed to be 0 when not provided for comparison purposes, however, if the underlying RPM DB entry does not have
+// an epoch specified it would be slightly disingenuous to display a value of 0.
+func toElVersion(metadata pkg.RpmdbMetadata) string {
```

Comment:
It's not clear to me what this function name (`toElVersion`) means... is this a domain-specific concept?

Reply:
The name derives from domain-specific concepts, yes.

RedHat has a "version naming scheme" of `name-version-release-arch` where the `version-release` is what is used to make comparisons within the RPM software --this is approach is shared with CentOS as well. You'll often see "rhel" which stands for RedHat Enterprise Linux or "el" which generically stands for Enterprise Linux and includes CentOS and similar flavors (basically means "*EL").

Seeing that this `version-release` pair is what is common to use as a package version for RHEL based distros, it made sense to name this function in a way that distinguishes it from just the name "toVersion", and "toVersionRelease" or "toVersionReleasePair" really didn't resonate / make sense. This is versioning scheme applies fairly universally within the EL ecosystem, so I ended up calling it "toElVersion" since this is the version value that can be compared within the EL ecosystem.

Reply:
Cool, this is helpful! Thanks!

I don't terribly mind the `VersionRelease` naming approach. Lately I've been working with Kubernetes code, and they have a struct that describes a resource's group, version, and kind (each as its own struct field). For the name of the struct, rather than invent a new term, they call it `GroupVersionKind`. I thought that was interesting... it was surprisingly straightforward.

But I see your point about the desire to harken back to the Enterprise Linux domain explicitly.

What about something like `toEnterpriseLinuxVersion`? It's longer than the acronym, but it's completely unambiguous.

A next step down in my mind would be `toELVersion`, to capitalize both letters of the acronym. This seems to be the convention in Go, to have all letters of the acronym uppercase if the first letter is uppercase. This acronym approach is less ideal though, from my perspective.

Reply:
I don't have strong opinions here but I don't think there is an _"EL ecosystem"_. "RedHat Enterprise Linux" is one thing as a whole. The problem does bubble up when packages use/advertise the `"el"` as part of their convention, and those packages can be seen in CentOS, RHEL, and others. 
"EL" is domain specific for sure, and if I had to pick I'd prefer `toELVersion`

Reply:
👍  renaming to `toELVersion`